doc/usage/generating_ssl_certificates.md

```diff
@@ -0,0 +1,140 @@
+        └── ssl
+            ├── ca
+            |   └── ca.crt
+            └── tls
+                ├── tls.crt
+                └── tls.key
+```
+
+<br>
+
+- `backup-restore` container
+
+```console
+/
+└── var
+    ├── etcd                      Contains the CA and server certs for etcd server
+    |   └── ssl
+    |       ├── ca
+    |       |   └── ca.crt
+    |       └── tls
+    |           ├── tls.crt
+    |           └── tls.key
+    └── etcdbr                    Contains the CA cert for etcd backup-restore server
+        └── ssl
+            └── ca
+                └── ca.crt
+```
+
+## Generating the certificates
+
+### Installing openssl
+
+```console
+# For Mac users
+brew install openssl
+
+# For other flavours of Unix
+apk install openssl
+
+mkdir openssl && cd openssl
+```
+
+### Generating certs for etcd server authentication
+
+#### Generating CA cert bundle
+
+```console
+openssl genrsa -out ca.key 2048
+openssl req -new -key ca.key -subj "/CN=etcd" -out ca.csr
+
+cat > ca.csr.conf <<EOF
+[ v3_ext ]
+keyUsage=critical,digitalSignature,keyEncipherment,keyCertSign,cRLSign
+basicConstraints=critical,CA:TRUE
+EOF
+
+openssl x509 -req -in ca.csr -signkey ca.key -out ca.crt -sha256 -days 3653 -extensions v3_ext -extfile ca.csr.conf
+
+# view contents of the generated certificate
+openssl x509 -in ca.crt -noout -text
+```
+
+#### Generating TLS key-pair
+
+```console
+openssl genrsa -out server.key 2048
+
+# In the `alt_names` section of server.csr.conf, replace all occurrences of `mynamespace` with the namespace into which you'll deploy the helm chart
+cat > server.csr.conf <<EOF
+[ req ]
+default_bits = 2048
+prompt = no
+default_md = sha256
+req_extensions = req_ext
+distinguished_name = dn
+
+[ dn ]
+CN = etcd-server
+
+[ req_ext ]
+subjectAltName = @alt_names
+
+[ alt_names ]
+DNS.1 = main-etcd-0
+DNS.2 = main-etcd-client
+DNS.3 = main-etcd-client.mynamespace
+DNS.4 = main-etcd-client.mynamespace.svc
+DNS.5 = main-etcd-client.mynamespace.svc.cluster.local
+
+[ v3_ext ]
+keyUsage=critical,digitalSignature,keyEncipherment
+extendedKeyUsage=serverAuth,clientAuth
+basicConstraints=critical,CA:FALSE
+subjectAltName=@alt_names
+EOF
+
+openssl req -new -key server.key -out server.csr -config server.csr.conf
+
+openssl x509 -req -in server.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out server.crt -sha256 -days 3653 -extensions v3_ext -extfile server.csr.conf
+
+# view contents of the generated certificate
+openssl x509 -in server.crt -noout -text
+```
+
+### Generating certs for etcdbr server authentication
+
+Follow the same steps as [generating certs for etcd](#generating-certs-for-etcd-server-authentication), but replace all occurrences of `etcd` with `etcdbr`. You will also need to add `localhost` to the SAN DNS list if you're deploying the etcd setup via the provided helm chart. If deploying by any other means, or if testing locally, please tweak the config accordingly.
```

Comment:
Just mention the `alt_names` section content separately for `etcdbr`. It's not clear enough.

Reply:
Done